src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteHandlers/DetectAmbiguousRoutes.cs

```diff
@@ -199,6 +214,28 @@ private static bool HasMatchingHttpMethods(ImmutableArray<string> httpMethods1,
             return false;
         }
 
+        private static bool AreArgumentsEqual(ImmutableArray<IArgumentOperation> arguments1, ImmutableArray<IArgumentOperation> arguments2)
+        {
+            if (arguments1.Length != arguments2.Length)
+            {
+                return false;
+            }
+
+            for (var i = 0; i < arguments1.Length; i++)
+            {
+                var arg1 = arguments1[i];
+                var arg2 = arguments2[i];
+
+                if (arg1?.Parameter?.Name != arg2?.Parameter?.Name ||
+                    !Equals(arg1?.Value.ConstantValue.Value, arg2?.Value.ConstantValue.Value))
```

Comment:
Is this valid? What if the argument doesn't have a `ConstantValue`? What if it's different variables? The parameter name (unlike the argument name) will always be `prefix` for two calls to `MapGroup`, right?

Even if we compared argument names instead of parameter names, it could still be an argument that was reassigned. It feels like this should only return true if the arguments are the exact same constant values instead of assuming any list of arguments of the same length are equal unless they are different constant values.